x-pack/plugins/observability/public/components/app/section/alerts/index.tsx

```diff
@@ -34,20 +38,64 @@ const allTypes = {
   }),
 };
 
-interface Props {
-  alerts: Alert[];
-}
-
-export function AlertsSection({ alerts }: Props) {
+export function AlertsSection() {
   const { config, core } = usePluginContext();
   const [filter, setFilter] = useState(ALL_TYPES);
   const manageLink = config.unsafe.alertingExperience.enabled
     ? core.http.basePath.prepend(paths.observability.alerts)
     : core.http.basePath.prepend(paths.management.rules);
-  const filterOptions = uniqBy(alerts, (alert) => alert.consumer).map(({ consumer }) => ({
-    value: consumer,
-    text: consumer,
-  }));
+
+  const { data, status } = useFetcher(() => {
+    return getObservabilityAlerts({ core });
+  }, [core]);
+
+  const alerts = useMemo(() => data || [], [data]);
+
+  const filterOptions = useMemo(() => {
+    if (!alerts) {
+      return [];
+    }
+    return uniqBy(alerts, (alert) => alert.consumer).map(({ consumer }) => ({
+      value: consumer,
+      text: consumer,
+    }));
+  }, [alerts]);
+
+  const isError = status === FETCH_STATUS.FAILURE;
+  const isLoading = status !== FETCH_STATUS.SUCCESS && !isError;
+
+  if (isLoading) {
+    return (
+      <EuiFlexGroup alignItems="center" justifyContent="center" responsive={false}>
+        <EuiFlexItem grow={false}>
+          <EuiLoadingSpinner size="xl" />
+        </EuiFlexItem>
+      </EuiFlexGroup>
+    );
+  }
+
+  if (isError) {
+    return (
+      <EuiFlexGroup alignItems="center" justifyContent="spaceBetween" responsive={false}>
+        <EuiFlexItem>
+          <EuiCallOut
+            title={i18n.translate('xpack.observability.overview.alert.errorTitle', {
+              defaultMessage: "We couldn't load the alert data",
+            })}
```

Comment:
Please move to a `translations` module

Reply:
I'm not sure I understand. You mean to take this out to a separate file?

Reply:
I think he means to move it to [this file](https://github.com/elastic/kibana/blob/main/x-pack/plugins/observability/public/config/translations.ts), but I'm not sure if these strings should be moved there.

What are the benefits of keeping the translations in one single place? I usually find this type of file difficult to maintain and I like to keep the translations inside the component. Not saying that I'm against it, but I would be happy to hear the reasoning behind that decision 😊

Reply:
I'm also not sure about the benefit of it. I feel it adds an extra layer of indirection.

@claudiopro could you elaborate on this?

Reply:
Sure, sorry for the lack of context! In RAC we use a layout where translation strings live in a `translations` module and importing them where needed as constants, e.g. `import * as i18n from './translations';`. Not sure if choice is project specific, so feel free to ignore if your project has a different approach. @mgiota can provide more background.